Patch: Restart WireGuard on WAN IP Change
Author: Andrew Collings
Date: 2025-06-12
Description: This patch triggers a WireGuard service restart when the WAN interface receives a new IP address to prevent sticking on lower-tier gateways.

--- a/src/etc/rc.newwanip	2025-02-24 15:04:58
+++ b/src/etc/rc.newwanip	2025-02-24 15:12:05
@@ -44,6 +44,9 @@
 	/* restart packages */
 	log_error("{$g['product_label']} package system has detected an IP change or dynamic WAN reconnection - $oldip ->  $curwanip - Restarting packages.");
 	send_event("service reload packages");
+	log_error("Restarting WireGuard after IP change.");
+	mwexec("/usr/local/etc/rc.d/wireguardd restart");
+	log_error("Restarting FRR after WG interface change.");
+ require_once('/usr/local/pkg/frr.inc');
+ frr_generate_config();
+ mwexec('/usr/local/etc/rc.d/frr.sh restart');
+
 }
 
 /* Interface IP address has changed */
